File: rs4/rs4补环境.py
import re
from lxml import etree
import execjs
import requests
from parsel import Selector

headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36"
}
url = "https://hd.ghzyj.sh.gov.cn/2017/xxgk/zcjd/"
response = requests.get(url, headers=headers, verify=False)
with open('./zcjd.html', 'w', encoding='utf8') as f:
    f.write(response.text)
ic = response.cookies.get("hd.ghzyj.sh.gov.cn_443_ic")  # 获取cookie
S80 = response.cookies.get("FSSBBIl1UgzbN7N443S")  # 获取cookie
print(f"80S：{S80}")
etree_html = etree.HTML(response.text)
content = etree_html.xpath("//meta/@content")[1]
print(f"content：{content}")
script = etree_html.xpath("//script")
js_url = 'https://hd.ghzyj.sh.gov.cn' + script[0].xpath("@src")[0]
resp = requests.get(js_url, headers=headers)
js_content = '''var content="contentfff";'''.replace("contentfff", content)
js_dfe1657 = resp.text
js_auto_code = script[1].xpath("text()")[0]
get_cookie = """
function get_cookie(){
    return document.cookie;
}
"""
with open("rs4补环境.js", encoding='utf-8') as f:
    js_env_code = f.read()
run_code = js_content + ";" + "\r\n" + js_env_code + ";" + "\r\n" + js_dfe1657 + ";" + "\r\n" + "\r\n" + "debugger;" + js_auto_code + ";" + "\r\n" + get_cookie + "\r\n"

with open('./run_coder.js', 'w', encoding='utf8') as f:
    f.write(run_code)

ru = execjs.compile(run_code)

cookie = ru.call("get_cookie").split("=")[1].split(";")[0]
print(f"生成的cookie：{cookie}")
# 防止滥用，发包逻辑已删除

File: rs4/rs4补环境.js
window = global;
top = window;
window.self = window;
Window = function Window(){}
Window.prototype = window.__proto__;

setTimeout = function () {return {}};
setInterval = function () {return{}};

document = {
    "cookie": '',
    "characterSet": 'UTF-8',
    "charset": "UTF-8",
    "scripts": ['script', 'script'],
};

navigator = {
    appCodeName: "Mozilla",
    appName: "Netscape",
    appVersion: "5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54",
    userAgent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54",
    vendor: "Google Inc.",
    webdriver: false,
    userActivation: { hasBeenActive: false, isActive: false },
    mimeTypes:[{"application/x-shockwave-flash":""}]
};

location = {
    "ancestorOrigins": {},
    "href": "https://hd.ghzyj.sh.gov.cn/2017/xxgk/zcjd/",
    "origin": "https://hd.ghzyj.sh.gov.cn",
    "protocol": "http:",
    "host": "hd.ghzyj.sh.gov.cn",
    "hostname": "hd.ghzyj.sh.gov.cn",
    "port": "",
    "pathname": "/2017/xxgk/zcjd/",
    "search": "",
    "hash": ""
};


div = {}
createElement = function () {
    console.log('创建对象：', arguments[0]);
    switch (arguments[0]) {
        case 'div':
            return div;
        default:
            return {};
    }
};
document.createElement = createElement;

getElementsByTagName = function () {
    console.log('获取元素：', arguments[0]);
    switch (arguments[0]) {
        case 'i':
            return [];
        default:
            return [];
    }
}
div.getElementsByTagName = getElementsByTagName;


getElementsByTagName = function () {
    console.log('获取元素：', arguments[0]);
    switch (arguments[0]) {
        case 'meta':
            return [{
                "content": content,
                "parentNode": {
                    removeChild: function () { }
                }
            }, {
                "content": content,
                "parentNode": {
                    removeChild: function () { }
                }
            }]
        default:
            return [];
    }
};
document.getElementsByTagName = getElementsByTagName;


document.documentElement = {"style":{}}

document.addEventListener = function () {
    console.log(arguments)
    return {}
};
document.attachEvent = function () { 
    console.log(arguments);
    return {} 
};

window.addEventListener = function () {
    console.log(arguments)
    return {}
};
window.attachEvent = function () { 
    console.log(arguments);
    return {} 
};
window.document = document;
window.navigator = navigator;
window.location = location;
delete __filename;
delete __dirname;


